django/contrib/admin/helpers.py

```diff
@@ -280,7 +280,9 @@ def fields(self):
             if not self.has_change_permission or field_name in self.readonly_fields:
                 yield {
                     'name': field_name,
-                    'label': meta_labels.get(field_name) or label_for_field(field_name, self.opts.model, self.opts),
+                    'label': meta_labels.get(field_name) or label_for_field(
+                        field_name, self.opts.model,
+                        self.opts, form=empty_form),
```

Comment:
It looks that this change is not required. All tests pass without it.

Reply:
I not write tests for this scenario. I'm not remember if have readonly field or has_change_permission on "add" in inline forms. I write this forgotten test.

Reply:
How I get your commit?

Reply:
`git pull - - rebase` on your branch should be enough.

Reply:
Readonly field must be a property or a model field, in both cases it works without this change. I added tests for inline without change permission.